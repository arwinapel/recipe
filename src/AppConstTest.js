const date = new Date();
const tomorrowDate = new Date(date);
tomorrowDate.setDate(tomorrowDate.getDate() + 1)
export const TOMORROW = `${tomorrowDate.getMonth() < 10 ? `0${tomorrowDate.getMonth() + 1}` : `${tomorrowDate.getMonth() + 1}`}/${tomorrowDate.getDate()}/${tomorrowDate.getFullYear()}`;
export const successIngredients = {
    data: [
        {
            "title": "Ham",
            "use-by": "2020-11-25"
        },
        {
            "title": "Cheese",
            "use-by": "2020-01-08"
        },
    ]
};
export const successRecipe = {
    data: [
        {
            "title": "Ham and Cheese Toastie",
            "ingredients": [
                "Ham",
                "Cheese",
                "Bread",
                "Butter"
            ]
        },
        {
            "title": "Salad",
            "ingredients": [
                "Lettuce",
                "Tomato",
                "Cucumber",
                "Beetroot",
                "Salad Dressing"
            ]
        },
        {
            "title": "Hotdog",
            "ingredients": [
                "Hotdog Bun",
                "Sausage",
                "Ketchup",
                "Mustard"
            ]
        }
    ]
};
export const addIngredient = jest.fn();
export const getRecipe = jest.fn();
export const ingredient = {
    title: 'Ham',
    useBy: '2019-12-12',
    addIngredient: addIngredient,
    selectedData: [],
};