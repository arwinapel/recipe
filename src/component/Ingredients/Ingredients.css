.ingredients-container {
    border: 1px solid #c7c7c7;
    border-radius: 0.3em;
    background-color: white;
    box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
    height: 80vh;
    overflow-y: auto;
}

.btn-get-recipe {
    margin-bottom: 0.5em !important;
}