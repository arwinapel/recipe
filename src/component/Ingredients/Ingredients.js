import React from "react";
import PropTypes from "prop-types";
import Ingredient from "../Ingredient/Ingredient";
import './Ingredients.css';
import List from '@material-ui/core/List';

function Ingredients({data, selectedData, addIngredient}) {
    const ingredients = data.map((el, idx) => (
        <Ingredient
            key={idx}
            selected={selectedData.includes(el.title)}
            title={el.title}
            useBy={el['use-by']}
            addIngredient={addIngredient}
        />
    ));

    return (
        <List
            dense
            disablePadding
            className={"ingredients-container"}
        >
            {ingredients}
        </List>
    )
}

Ingredients.propTypes = {
    data: PropTypes.array.isRequired,
    selectedData: PropTypes.array.isRequired,
    addIngredient: PropTypes.func.isRequired,
};

export default Ingredients