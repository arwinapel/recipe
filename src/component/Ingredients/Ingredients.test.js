import React from 'react';
import {cleanup, render, fireEvent} from '@testing-library/react';
import Ingredients from "./Ingredients";
import {addIngredient, getRecipe, successIngredients} from "../../AppConstTest";

describe("Ingredients", () => {
    afterEach(cleanup)

    it("Render ingredients", () => {
        const {asFragment, getAllByTestId} = render(<Ingredients data={successIngredients.data}
                                                                 selectedData={[]}
                                                                 addIngredient={addIngredient}/>);
        const resultIngredients = getAllByTestId('ingredient').map(ingredient => ingredient.textContent);
        expect(resultIngredients.length).toBe(2);
        expect(asFragment).toMatchSnapshot();
    });
});