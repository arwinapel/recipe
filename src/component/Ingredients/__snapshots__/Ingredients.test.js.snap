// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients Render ingredients 1`] = `[Function]`;
